crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -4611,16 +4622,24 @@ impl<'db> TypeInferenceBuilder<'db> {
         // (2) We must *not* call `self.extend()` on the result of the type inference,
         //     because `ScopedExpressionId`s are only meaningful within their own scope, so
         //     we'd add types for random wrong expressions in the current scope
-        let iterable_type = if comprehension.is_first() {
+        let iterable_type = if comprehension.is_first()
+            && !target.is_attribute_expr()
+            && !target.is_subscript_expr()
```

Comment:
Should we use a positive test for names here? (in case the list of possibilities will ever be extended? or is that not possble?)